src/workers/postUpdated.ts

```diff
@@ -44,26 +51,29 @@ const worker: Worker = {
     const data: Data = messageToJson(message);
     logger.info({ data }, 'content-updated received');
     try {
-      const { post_id, updated_at } = data;
-      if (!post_id) {
-        return;
-      }
-      const updatedDate = new Date(updated_at);
+      // See if we received any rejections
+      const { reject_reason, post_id, updated_at, submission_id } = data;
       await con.transaction(async (entityManager) => {
-        // For now, we only allow Squad posts to be updated through this flow
-        const databasePost = await entityManager
-          .getRepository(ArticlePost)
-          .findOneBy({ id: post_id, origin: PostOrigin.Squad });
+        if (reject_reason) {
+          // Check if we have a submission id, we need to notify
+          if (!submission_id) {
+            logger.info({ data }, 'received rejection without submission id');
+          }
 
-        if (data?.origin === PostOrigin.Squad) {
-          data.source_id = UNKNOWN_SOURCE;
-        }
-
-        if (
-          !databasePost ||
-          databasePost.metadataChangedAt.toISOString() >=
-            updatedDate.toISOString()
-        ) {
+          const submissionRepo = con.getRepository(Submission);
+          const submission = await submissionRepo.findOneBy({
+            id: submission_id,
+          });
+          if (submission.status === SubmissionStatus.Started) {
```

Comment:
Just in case the submission doesn't exist

```suggestion
          if (submission?.status === SubmissionStatus.Started) {
```